timeline-chart/src/time-graph-unit-controller.ts

```diff
@@ -91,7 +95,15 @@ export class TimeGraphUnitController {
         if (this._viewRange.end > this.absoluteRange) {
             this._viewRange.end = this.absoluteRange;
         }
+        const newViewRangeLength = this.viewRangeLength;
+        this._scaleFactor = Number(prevViewRangeLength) / Number(newViewRangeLength);
```

Comment:
There is a property in the state controller called `zoomFactor` which has change handlers:
$$ZoomFactor = ZF = CanvasDisplayWidth/ViewRangeLength$$
The zoomFactor is updated every change to unitController.viewRange.  Could we move the logic for _scaleFactor into the stateController and move the handlers to onZoomChanged?  It would make more sense to only update the scale when the zoom changes.

Assuming canvasDisplayWidth doesn't change: the math works out very similarly:
In unit controller: $$scaleFactor = SF_{uc} = VR_{prev} / VR_{new}$$
In state controller: $$SF_{sc} = ZF_{prev} / ZF_{new} = (CDW_{prev}/VR_{prev}) /(CDW_{new}/VR_{new})$$
Since canvas display width won't change at the same time as view range:
$$CDW_{prev} = CDW_{new} \therefore SF_{sc} = SF_{uc}^{-1} = VR_{new} / VR_{prev}$$

Reply:
Also, we should probably add a handler to recalculate the zoom factor when the window resizes if it isn't already there.

Reply:
Hi William, thank you for your review. I moved everything into the state controller as you have suggested. 

> Could we move the logic for _scaleFactor into the stateController and move the handlers to onZoomChanged?

There are 2 cases right now that the scaleFactor needs to be changed: (1) When the view range changes (which will also corresponds to the zoom change), and (2) when the display canvas width changes (when user resize the timeline chart). Unfortunately, the zoom change event is not raised in the second case, so I cannot put everything in the handler for onZoomChanged (to the best of my knowledge). Please feel free to suggest/correct me if I miss something :).